Add setting to make vanished players invulnerable

MrNavaStar opened the pull request:
This is useful if vanished players are not in creative mode and you want to prevent them from getting blown up.
- I use this for a more restricted spectator mode

Let me know if anything needs adjusting, Cheers!

DrexHD commented:
Thanks, I have changed the default to `false` to keep current behavior. I have also rewritten the invulnerability part to be more compatible with other mods that might update the value. Also to make sure stuff like, people being stuck in invulnerability, if the config is changed while they are vanished doesn't happen!

MrNavaStar commented:
> Thanks, I have changed the default to `false` to keep current behavior. I have also rewritten the invulnerability part to be more compatible with other mods that might update the value. Also to make sure stuff like, people being stuck in invulnerability, if the config is changed while they are vanished doesn't happen!

Ah I had thought about the getting stuck as invulnerable, but I wanted to wait to see your response. Glad you fixed it, and thanks for merging!
